README.md

```diff
@@ -2,134 +2,102 @@
 
 ### Enable/Disable zipkin tracing
 
-Zipkin relies on CrossApplicationTracer library (https://github.com/mdsol/Medidata.CrossApplicationTracer)'s TraceProvider to retrieve ids which are passed through the http request headers.
-
-Zipkin will record traces if TraceProvider's IsSampled is true.  
-
+Zipkin will record traces if IsSampled HTTP header is true.  
 This will happen if :
+- **a)** the caller of the app has set the IsSampled HTTP header value to true.
+- **OR**
+- **b)** the url request is not in the `ExcludedPathList` of ZipkinConfig , and using the `SampleRate`, it will determine whether or not to trace this request. `SampleRate` is the approximate percentage of traces being recorded in zipkin.
 
-a) the caller of the app has set the IsSampled http header value to true.
-
-OR
+## Configurations
+Please use `ZipkinConig` class to configure the module and verify these values and modify them according to your service/environment.
 
-b) the url request is not in the mAuthWhitelist appsetting config, and using the zipkinSampleRate, CrossApplicationTracer will determine whether or not to trace this request. zipkinSampleRate is the approximate percentage of traces being recorded in zipkin.
+- `Enable` - true: enable ZipkinMiddleware, false: disable ZipkinMiddleware.
+- `Domain` - is a valid public facing base url for your app instance.
+- `ZipkinBaseUri` - is the zipkin scribe/collector server URI with port to send the Spans
+- `ServiceName` - name of your Service that zipkin will use to label the trace
+- `SpanProcessorBatchSize` - how many Spans should be sent to the zipkin scribe/collector in one go.
+- `SampleRate` - 1 decimal point float value between 0 and 1. This value will determine randomly if the current request will be traced or not.	 
+- `NotToBeDisplayedDomainList`(optional) - It will be used when logging host name by excluding these strings in service name attribute
+	e.g. domain: ".xyz.com", host: "abc.xyz.com" will be logged as "abc" only    
+- `ExcludedPathList`(optional) - Path list that is not needed for tracing. Each item must start with "/". 
 
-### Configurations
-Below are the configurations that are needed.  
-
-1) appsettings.config
-
-Add the below additional configurations. Please verify these values and modify them according to your service/environment.
 
 ```
-<appSettings>
-  <add key="domain" value="http://application-sandbox.xyz.com" />
-  <add key="zipkinBaseUri" value="http://zipkinvm.cloudapp.net:9411" />
-  <add key="zipkinServiceName" value="Name of your Service i.e. MyApplication" />
-  <add key="zipkinSpanProcessorBatchSize" value="10" />
-  <add key="zipkinSampleRate" value="0.5" />
-  <add key="zipkinNotToBeDisplayedDomainList" value=".xyz.com,.myApplication.net" />
-  <add key="zipkinExcludedUriList" value="/check_uri,/status" />
-</appSettings>
+var config = new ZipkinConfig
+{
+	Enable = true,
+	Domain = "service.com",
```

Comment:
The value of domain is going to be used inside constructor of Uri and needs to be a valid uri string
Please check: https://github.com/mdsol/Medidata.ZipkinTracerModule/blob/feature/middleware/src/Medidata.ZipkinTracer.Core/SpanTracer.cs#L129

This example would create exception. We might need to change this to
```
Domain = "https://service.com",
```